File: gabe/climbing_stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n == 1 || n == 2) return n;
        
        vector<int> stairsOf(n);
        
        stairsOf[0] = 1;
        stairsOf[1] = 2;
        
        for(int i = 2; i < n; ++i) {
            stairsOf[i] = stairsOf[i-1] + stairsOf[i-2];
        }
        
        return stairsOf[n-1];
        
    }
};

File: brady/climbing_stairs.java
class Solution {
  
    // O(n) time, O(1) space
    public int climbStairs(int n) {
        // Edge cases for small n
        if (n < 2) return 1;

        // Recurrence:
        // solutions[0] = 1
        // solutions[1] = 1 + solutions[0] = 2
        // solutions[i] = solutions[i-1] + solutions[i-2]
        // We can use the fibonacci sequence starting from values 1 and 2
        int fib1 = 1;
        int fib2 = 2;
        for (int i = 2; i < n; ++i) {
            int nextVal = fib1 + fib2;
            fib1 = fib2;
            fib2 = nextVal;
        }

        return fib2;
    }
}
